### DISCUSSION TITLE
Adding Meshes for Field Data Sharing

### DISCUSSION THREAD

**Discussion by JhonCordova** [AMReX-Combustion/PeleC#755] [General]
I'd like to incorporate 2 additional meshes into PeleC and subsequently enable them to share information from the solved fields, i.e., velocity field. Does anyone have suggestions on which parts of the code I should begin modifying or what factors I should take into account?

**Comment by baperry2** [comment]
If I understand correctly, you basically want two separate PeleC simulations running at the same time on different grids that communicate data between each other on-the-fly?

The [AMReX MultiBlock](https://github.com/AMReX-Codes/amrex/tree/development/Tests/MultiBlock) framework lets you pass data between meshes of AMReX codes in memory for runtime coupling of simulations on different grids. If you want to incorporate multiple separate meshes in a single Pele simulation, that's the route I would go. That would be cleanest, most efficient, and perhaps simplest way of doing this, but it would likely still be very involved. Off the bat, it might be nontrivial to just to let two PeleC objects successfully coexist, especially if configured differently, before even considering having them communicate. But if you could pull it off, it would be very cool!

If you want to go this route, you can look at at ERF (started as a fork of PeleC, now something very different), where we've played with MultiBlock for [ERF/ERF](https://github.com/erf-model/ERF/tree/development/Exec/DevTests/MultiBlock) coupling and [ERF/amr-wind](https://github.com/erf-model/erf-amrwind-driver) coupling.

**Reply by JhonCordova** [reply]
Essentially, I aim to solve the same problem with identical settings, but utilizing three different grids (grid 1, 2, and 3 with an higher number of cells in the x, y, z directions, respectively). I'm leaning towards achieving this within a single PeleC simulation. Thus, I'm thinking about creating multiple AMR objects that can communicate seamlessly. Exploring the AMReX MultiBlock framework more extensively is part of my plan. Any additional suggestions you may have would be highly appreciated.
Thanks
